cobalt/browser/h5vcc_metrics/public/mojom/h5vcc_metrics.mojom

```diff
@@ -19,7 +19,7 @@ interface MetricsListener {
 };
 
 interface H5vccMetrics {
-  AddListener(pending_remote<MetricsListener> listener) => ();
+  AddListener(pending_remote<MetricsListener> listener);
```

Comment:
I'm still confused. For backwards compatibility reasons, doesn't this need to match the existing naming of the current H5vccMetrics API in Cobalt 25? This method is `onMetricEvent` there: https://source.corp.google.com/h/github/youtube/cobalt/+/24.lts.1+:cobalt/h5vcc/h5vcc_metrics.idl;l=32;drc=0ad4522dd6a585a43cb5b11108a8e0ffc3d80d2c

Reply:
This is the mojom file, which defines the IPC communication between Blink and the Browser; it's an implementation detail and not public to JS (Web) clients, so it could be called anything we want. The IDL file names are the ones that Kabuki (or other WebApps) would care about and I tried to keep those unmodified (except for the event, which is now an Event).

Reply:
It's apparently difficult for us to maintain that same API and Kabuki has said they can change the callsites of these callback APIs with relative ease

Reply:
Thanks. Ideally, we keep this API name the same on the JS side. There's no real need to change it. It is relatively straightforward to support multiple API definitions, but it is not free and it makes it unnecessarily complex. The current JS logic  would need to branch on Cobalt version being >= 26 to decide which API to use, as would the unit tests.

Reply:
If there is a good way to maintain the same API, I agree that would be preferable. After talking with Colin about onDeepLink it seemed infeasible, but @yell0wd0g if you know of a way to do it, then great.

Reply:
> The current JS logic would need to branch on Cobalt version being >= 26 to decide which API to use, as would the unit tests.

Just to be sure, this Mojom file is _not_ surfaced to the Web. This AddListener is never going to be surfaced to JS. 

The JS API is in h_5_vcc_metrics.idl further below.

Aside from that, the Mojom methods below correspond to IDL ones 1-to-1, which I agree simplifies code understanding and maintnenace. But there's no IDL equivalent to register a listener, because that's all taken care of by Web scaffolding, so this AddListener is a net Mojo addition, and I called it so to resemble similar patterns elsewhere.

Reply:
> Just to be sure, this Mojom file is not surfaced to the Web. This AddListener is never going to be surfaced to JS.

Yes, sorry, I understand this. I was trying to respond to both you and Andrew in a single comment. I'm realizing now though it might be more todo with maintaining the mojo=>idl parity that he was referring to. Carry on!

Reply:
Marking this as Fixed, please reopen if there's still concerns.